Mathlib/MeasureTheory/Constructions/EventuallyMeasurable.lean

```diff
@@ -0,0 +1,117 @@
+/-
+Copyright (c) 2024 Felix Weilacher. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Felix Weilacher
+-/
+
+import Mathlib.MeasureTheory.MeasurableSpace.Defs
+import Mathlib.Order.Filter.CountableInter
+
+/-!
+# Measurability modulo a filter
+
+In this file we consider the general notion of measurability modulo a σ-filter.
+Two important instances of this construction are null-measurability with respect to a measure,
+where the filter is the collection of co-null sets, and
+Baire-measurability with respect to a topology,
+where the filter is the collection of comeager (residual) sets.
+TODO: Implement the latter.
+
+## Main definitions
+
+* `EventuallyMeasurableSpace`: A `MeasurableSpace` on a type `α` consisting of sets which are
+  `Filter.EventuallyEq` to a measurable set with respect to a given `CountableInterFilter` on `α`
+  and `MeasurableSpace` on `α`.
+* `EventuallyMeasurableSet`: A `Prop` for sets which are measurable with respect to some
+  `EventuallyMeasurableSpace`.
+* `EventuallyMeasurable`: A `Prop` for functions which are measurable with respect to some
+  `EventuallyMeasurableSpace` on the domain.
+
+-/
+
+open Filter Set MeasurableSpace
+
+variable {α : Type*} (m : MeasurableSpace α) (l : Filter α) [CountableInterFilter l] {s t: Set α}
+
+/-- The `MeasurableSpace` of sets which are measurable with respect to a given σ-algebra `m`
+on `α`, modulo a given σ-filter `l` on `α`. -/
+def EventuallyMeasurableSpace : MeasurableSpace α where
+  MeasurableSet' s := ∃ t, MeasurableSet t ∧ s =ᶠ[l] t
+  measurableSet_empty := ⟨∅, MeasurableSet.empty, EventuallyEq.refl _ _ ⟩
+  measurableSet_compl := fun s ⟨t, ht, hts⟩ => ⟨tᶜ, ht.compl, hts.compl⟩
+  measurableSet_iUnion s hs := by
+    choose t ht hts using hs
+    exact ⟨⋃ i, t i, MeasurableSet.iUnion ht, EventuallyEq.countable_iUnion hts⟩
+
+/-- We say a set `s` is an `EventuallyMeasurableSet` with respect to a given
+σ-algebra `m` and σ-filter `l` if it differs from a set in `m` by a set in
+the dual ideal of `l`. -/
+def EventuallyMeasurableSet (s : Set α) : Prop := @MeasurableSet _ (EventuallyMeasurableSpace m l) s
+
+variable {l m}
+
+theorem MeasurableSet.eventuallyMeasurableSet (hs : MeasurableSet s) :
+    EventuallyMeasurableSet m l s :=
+  ⟨s, hs, EventuallyEq.refl _ _⟩
+
+theorem EventuallyMeasurableSpace.measurable_le : m ≤ EventuallyMeasurableSpace m l :=
+  fun _ hs => hs.eventuallyMeasurableSet
+
+theorem eventuallyMeasurableSet_of_mem_filter (hs : s ∈ l) : EventuallyMeasurableSet m l s :=
+  ⟨univ, MeasurableSet.univ, eventuallyEq_univ.mpr hs⟩
+
+/-- A set which is `EventuallyEq` to an `EventuallyMeasurableSet`
+is an `EventuallyMeasurableSet`. -/
+theorem EventuallyMeasurableSet.eventuallyMeasurableSet_of_eventuallyEq
```

Comment:
```suggestion
theorem EventuallyMeasurableSet.congr
```
It's probably a good idea to follow the naming convention already used for null measurable sets.